agent/agent.go

```diff
@@ -3344,6 +3347,28 @@ func (a *Agent) ReloadConfig(newCfg *config.RuntimeConfig) error {
 	return nil
 }
 
+func (a *Agent) cleanSidecars() {
+	for range time.Tick(5 * time.Minute) {
```

Comment:
This goroutine will leak when agent is stopped which will mean thousands of them running during test suite.

Reply:
Fixed